usr/share/rear/prep/GNU/Linux/305_include_s390_tools.sh

```diff
@@ -0,0 +1,36 @@
+#
+#  s390 zIPL boot loader and grubby for configuring boot loader`
+
+[ "$ARCH" == "Linux-s390"  ] || return 0
+
+test -d $VAR_DIR/recovery || mkdir -p $VAR_DIR/recovery
+
+local bootdir="$( echo -n /boot/ )"
+test -d "$bootdir" || $bootdir='/boot/'
+
+# cf. https://github.com/rear/rear/issues/2137
+# findmnt is used the same as grub-probe to find the device where /boot is mounted
+# example
+# findmnt -no SOURCE --target /boot
+# --> /dev/dasda1
+#
+# on sles:
+#   findmnt returns --> /dev/dasda3[/@/.snapshots/1/snapshot]
+#   use 300_include_grub_tools.sh instead of this file (grub2-probe)
+if has_binary findmnt ; then
+    findmnt -no SOURCE --target $bootdir >$VAR_DIR/recovery/bootdisk || return 0
+fi
+
+# Missing programs in the PROGS array are ignored:
+# zipl and grubby are  added in conf/Linux-s390x.conf
+PROGS=( "${PROGS[@]}" findmnt dasd_cio_free dasdfmt dasdinfo dasdstat dasdview dasdconf.sh fdasd qetharp qethconf qethqoat )
+PROGS=( "${PROGS[@]}" cmsfscat  cmsfsck  cmsfscp  cmsfslst  cmsfsvol )
+PROGS=( "${PROGS[@]}" chccwdev chshut chiucvallow chchp tape390_crypt tape390_display )
+PROGS=( "${PROGS[@]}" cmsfscat  cmsfsck  cmsfscp  cmsfslst  cmsfsvol )
```

Comment:
The line
```
PROGS=( "${PROGS[@]}" cmsfscat  cmsfsck  cmsfscp  cmsfslst  cmsfsvol )
```
is there needlessly two times.

Reply:
In general to add elements to an array I personally prefer using `+=` like
```
PROGS+=( cmsfscat cmsfsck cmsfscp cmsfslst cmsfsvol )
```
because it looks simpler and makes it more clear what the intent is.

I tested that `arr+=( additional elements )`
works down to SLES10 with GNU bash version 3.1.17
so it should be safe to use it in nowadays ReaR.

Reply:
thank you,  i removed the dup and changed to the newer syntax